apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/settings/update-key-remaining.tsx

```diff
@@ -176,37 +150,11 @@ export const UpdateKeyRemaining: React.FC<Props> = ({ apiKey }) => {
                   </FormItem>
                 )}
               />
-
-              <FormField
-                control={form.control}
-                name="refill.interval"
-                render={({ field }) => (
-                  <FormItem className="">
-                    <FormLabel>Refill Rate</FormLabel>
-                    <Select
-                      onValueChange={field.onChange}
-                      defaultValue="none"
-                      value={field.value}
-                      disabled={!form.watch("limitEnabled") === true}
-                    >
-                      <SelectTrigger>
-                        <SelectValue />
-                      </SelectTrigger>
-                      <SelectContent>
-                        <SelectItem value="none">None</SelectItem>
-                        <SelectItem value="daily">Daily</SelectItem>
-                        <SelectItem value="monthly">Monthly</SelectItem>
-                      </SelectContent>
-                    </Select>
-                    <FormMessage />
-                  </FormItem>
-                )}
-              />
               <FormField
                 control={form.control}
                 disabled={
-                  form.watch("refill.interval") === "none" ||
-                  form.watch("refill.interval") === undefined ||
+                  form.watch("remaining") === undefined ||
+                  form.watch("remaining") === null ||
```

Comment:
If remaining being `0` is not a concern let's just do `!form.watch("remaining")`